core/src/main/scala/com/microsoft/azure/synapse/ml/fabric/FabricClient.scala

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.fabric
+
+import com.microsoft.azure.synapse.ml.logging.common.PlatformDetails
+import org.apache.log4j.Logger
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.JsValue
+
+import java.util.UUID
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+import java.net.URL
+
+object FabricClient extends RESTUtils {
+  private val PbiGlobalServiceEndpoints = Map(
+    "public" -> "https://api.powerbi.com/",
+    "fairfax" -> "https://api.powerbigov.us",
+    "mooncake" -> "https://api.powerbi.cn",
+    "blackforest" -> "https://app.powerbi.de",
+    "msit" -> "https://api.powerbi.com/",
+    "prod" -> "https://api.powerbi.com/",
+    "int3" -> "https://biazure-int-edog-redirect.analysis-df.windows.net/",
+    "dxt" -> "https://powerbistagingapi.analysis.windows.net/",
+    "edog" -> "https://biazure-int-edog-redirect.analysis-df.windows.net/",
+    "dev" -> "https://onebox-redirect.analysis.windows-int.net/",
+    "console" -> "http://localhost:5001/",
+    "daily" -> "https://dailyapi.powerbi.com/")
+
+  var CapacityID = "";
+  var WorkspaceID = "";
+  var ArtifactID = "";
+  var PbiEnv = "";
+  var FabricContext: Map[String, String] = Map[String, String]();
```

Comment:
maybe this var can instead be a lazy val which would triger the initialization logic